packages/hardhat-kadena/src/utils.ts

```diff
@@ -100,14 +129,56 @@ export const deployContractOnChains: DeployContractOnChains = async ({
         signer: contractDeployer,
         ...factoryOptions,
       });
-      const contract = await factory.deploy(
-        ...(overrides ? [...constructorArgs, overrides] : constructorArgs),
-      );
-      const deploymentTx = contract.deploymentTransaction();
-      if (!deploymentTx) {
-        throw new Error('Deployment transaction failed');
+
+      let contract;
+
+      // Use CREATE2 factory for deployment if salt is provided.
```

Comment:
I don't agree with this change. Now there is no way for devs to deploy using CREATE.  We should only be modifying the deployOnChainsUsingCreate2 to use the create2Factory system contract, not modifying deployContractOnChains to deploy using CREATE2